### PR TITLE
Add XGboost doc for training

### PR DESCRIPTION
**Pull request by johnugeorge**
Related: #2915

### CODE REVIEW

File: content/en/docs/components/training/xgboost.md [hunk 1]
```diff
@@ -0,0 +1,144 @@
++++
+title = "XGBoost Training"
+description = "Instructions for using XGBoost"
+weight = 30
+                    
++++
+
+{{% stable-status %}}
+
+This page describes XGBoostJob for training a machine learning model with XGBoost.
```
Comment (andreyvelich) [inline content/en/docs/components/training/xgboost.md:10]: Should we also add here:
```suggestion
This page describes `XGBoostJob` for training a machine learning model with XGBoost.
`XGBoostJob` is a Kubernetes [custom resource](https://kubernetes.io/docs/concepts/extend-kubernetes/api-extension/custom-resources/) to run XGBoost training jobs on Kubernetes. The Kubeflow implementation of `XGBoostJob` is in [`training-operator`](https://github.com/kubeflow/tf-operator).
```

File: content/en/docs/components/training/xgboost.md [hunk 2]
```diff
@@ -0,0 +1,144 @@
++++
+title = "XGBoost Training"
+description = "Instructions for using XGBoost"
```
Comment (andreyvelich) [inline content/en/docs/components/training/xgboost.md:3]:
```suggestion
title = "XGBoost Training (XGBoostJob)"
description = "Using XGBoostJob to train a model with XGBoost"
```

File: content/en/docs/components/training/xgboost.md [hunk 3]
```diff
@@ -0,0 +1,144 @@
...
+```
+
+
+## Creating a XGBoost Job
+
+You can create XGBoost Job by defining a XGboostJob config file. See the manifests for the [IRIS example](https://github.com/kubeflow/tf-operator/blob/master/examples/xgboost/xgboostjob.yaml). You may change the config file based on your requirements. eg: add `CleanPodPolicy` in Spec to `None` to retain pods after job termination.
+
+```
+cat xgboostjob.yaml
+```
```
Comment (andreyvelich) [inline content/en/docs/components/training/xgboost.md:41]: I think we can remove this

File: content/en/docs/components/training/xgboost.md [hunk 4]
```diff
@@ -0,0 +1,144 @@
...
+Deploy the XGBoostJob resource to start training:
+
+```
+kubectl create -f xgboostjob.yaml
```
Comment (andreyvelich) [inline content/en/docs/components/training/xgboost.md:45]: @johnugeorge What do you think about using GitHub link here as I mentioned in this PR: https://github.com/kubeflow/website/pull/2920#issue-728146538
Then, users don't need to copy the example.

**Review by terrytangyuan — APPROVED** [review]
Thanks!

/lgtm

**Pull request merged**